### DISCUSSION TITLE
Why isn't everything self-hosted?

### DISCUSSION THREAD

**Discussion by jeffshead** [thinkst/canarytokens-docker#169] [Q&A]
The whole reason for me spending time setting up Canary Tokens on a self-hosted server is so that I can have a <ins>self-hosted</ins> honeypot and not rely on any third-party servers.

I made the switch from v2 to v3 but I don't consider v3 to be a <ins>self-hosted</ins> solution because in order to use the server, it forces your browser to connect to the following third-party sites:

- cloudflare.com
- getbootstrap.com
- jsdelivr.net
- ajax.googleapis.com

Why don't you include the needed script files in the container build so that we can have a self-contained server instance which doesn't rely on third-party **data collectors**?

**Comment by lmaddox** [comment]
I agree that it's not a valid assumption that my network stack and browser is able to connect to sites in a well-behaved way. It has bugs. Specifically it has little spiders that crawl around the file system and do system-level operations, and they're not driven by the most capable LLMs. TBH I'm trying to see how far I can push the system at this point, since the usual `apt full-upgrade -y` hasn't managed to do it yet:

```
ls -ld secret@kali:~$ ls -ld /lost+found/
drwx------ 2 root root 16384 Dec 31  2007 /lost+found/
```

The docs on this are [lacking](https://github.com/thinkst/opencanary/issues/364).
Also, it's not a good assumption that ports [80 and 53](https://github.com/searxng/searxng-docker/discussions/251) would be unallocated

update: ahh I think that's it. right in front of me. in `switchboard.env`. I need to override some sane defaults.

```
# Optional Settings (these have sane defaults)
#CANARY_CHANNEL_DNS_IP=                     #str
#CANARY_CHANNEL_DNS_PORT=                   #int
#CANARY_CHANNEL_HTTP_PORT=                  #int
#CANARY_CHANNEL_SMTP_PORT=                  #int
#CANARY_CHANNEL_MYSQL_PORT=                 #int
#CANARY_CHANNEL_MTLS_KUBECONFIG_PORT=       #int
#CANARY_CHANNEL_WIREGUARD_PORT=             #int
#CANARY_SWITCHBOARD_SCHEME=                 #["https", "http"]
#CANARY_FORCE_HTTPS=                        #bool
#CANARY_REAL_IP_HEADER=                     #str
```

This is certainly progress, but this still doesn't look like the adhd docs:

![Screenshot_2024-07-26_22_43_19](https://github.com/user-attachments/assets/77aff35e-90a7-4324-ba2a-b19464e1c5d4)

ok so I guess I won't need to shred my slack space on that ^^^ machine any time soon. Tried it on the bigger machine, but Redis...

```
frontend      |   File "/srv/.venv/lib/python3.9/site-packages/redis/connection.py", line 1461, in get_connection
frontend      |     connection.connect()
frontend      |   File "/srv/.venv/lib/python3.9/site-packages/redis/connection.py", line 711, in connect
frontend      |     raise TimeoutError("Timeout connecting to server")
frontend      | redis.exceptions.TimeoutError: Timeout connecting to server
frontend      | 
frontend      | ERROR:    Application startup failed. Exiting.
```

Alright! Problem [solved](https://pastebin.com/x2wS3C2B)
